app/services/foreman_deprecation.rb

```diff
@@ -0,0 +1,24 @@
+class ForemanDeprecation
+  def self.deprecation_warning(moved_to_controller, moved_to_method, foreman_version_deadline, additional_info = nil,callstack = caller)
```

Comment:
Just committed, the changes to templates:

* https://github.com/theforeman/foreman/commit/510d53cd899134b54a935cdd675d69194a1f08bd#diff-72254ab53eb46e9862847d236566cb18R97
* https://github.com/theforeman/foreman/commit/510d53cd899134b54a935cdd675d69194a1f08bd#diff-ecca336ea4c708b230851d9b6e986de7R68
* https://github.com/theforeman/foreman/commit/510d53cd899134b54a935cdd675d69194a1f08bd#diff-fa8faf553918630d1afb4537fb180e90R4